### PR TITLE
Add katib controller flags to developers guide

### PR DESCRIPTION
**Pull request by annajung**
Signed-off-by: Anna Jung (VMware) <antheaj@vmware.com>

**What this PR does / why we need it**:
- Document Katib controller flags in the developer guide
- Flags were based on https://github.com/kubeflow/katib/blob/fe64c18172ba370fb3934552c50442f1da21ec1a/cmd/katib-controller/v1beta1/main.go#L50

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubeflow/katib/issues/1239

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.
N/A

**Release note**:

```release-note
NONE
```

/hold for feedback

### CODE REVIEW

**Review by gaocegege — COMMENTED** [review]
/lgtm

Thanks for your contribution! :tada: :+1:

File: docs/developer-guide.md [hunk 1]
```diff
@@ -60,6 +61,21 @@ You can update necessary files as follows:
...
+| ------------------------------- | ----------- | ------------------- | ----------------------------------------------------------------------------------------------------------------------- |
+| cert-localfs                    | bool        | false               | Store the webhook cert in local file system                                                                             |
+| enable-grpc-probe-in-suggestion | bool        | true                | Enable grpc probe in suggestions                                                                                        |
+| experiment-suggestion-name      | string      | "default"           | The implementation of suggestion interface in experiment controller                                                     |
+| metrics-addr                    | string      | ":8080"             | The address the metric endpoint binds to                                                                                |
+| trial-resources                 | value       | null                | The list of resources that can be used as trial template, in the form: Kind.version.group (e.g. TFJob.v1.kubeflow.org)  |
```
Comment (andreyvelich) [inline docs/developer-guide.md:74]: I think we can define the type here, WDYT @annajung @gaocegege 
```suggestion
| trial-resources                 | []schema.GroupVersionKind       | null                | The list of resources that can be used as trial template, in the form: Kind.version.group (e.g. TFJob.v1.kubeflow.org)  |
```

**Review by andreyvelich — COMMENTED** [review]
Thank you to update this @annajung!
I left one comment.

File: docs/developer-guide.md (hunk 1 above)
Comment (annajung) [reply docs/developer-guide.md:74]: That sounds good to me! Updated with the suggestion

**Comment by andreyvelich** [comment]
Thank you for the contribution!
/lgtm
/approve

**Pull request merged**